src/RoslynPad/UserSettingsDialog.xaml.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            PerformSave();
+            RestartApplication();
+            Close();
+        }
+
+        private void Cancel_Click(object sender, RoutedEventArgs e)
+        {
+            Close();
+        }
+
+        internal static string GetUserDocumentPath()
+        {
+            var userDefinedPath = Properties.Settings.Default.DocumentPath;
+            return !string.IsNullOrEmpty(userDefinedPath) && System.IO.Directory.Exists(userDefinedPath)
+                ? userDefinedPath
+                : GetDefaultDocumentPath();
+        }
+
+        private static string GetDefaultDocumentPath()
+        {
+            return System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "RoslynPad");
+        }
+
+        private void RestartApplication()
+        {
+            // TODO: Review use of System.Windows.Forms reference.
+            System.Windows.Forms.Application.Restart();
+            Environment.Exit(0);
+        }
+
+        public void Show()
+        {
+            _dialog = new InlineModalDialog
+            {
+                Owner = Application.Current.MainWindow,
+                Content = this
+            };
+            _dialog.Show();
+        }
+
+        public void Close()
+        {
+            _dialog?.Close();
+        }
+
+        private void PerformSave()
+        {
+            SaveDocumentPath(UserDocumentPath);
+            SaveCreateSamples(UserCreateSampleFiles);
+            Result = SaveResult.Save;
+        }
+
+        private void SaveDocumentPath(string documentPath)
+        {
+            if (Directory.Exists(documentPath))
+            {
+                Properties.Settings.Default.DocumentPath = documentPath;
+                Properties.Settings.Default.Save();
+            }
+        }
+
+        private void SaveCreateSamples(bool createSamples)
+        {
+            Properties.Settings.Default.CreateSamples = createSamples;
+            Properties.Settings.Default.Save();
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        [NotifyPropertyChangedInvocator]
+        protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        [NotifyPropertyChangedInvocator]
+        protected bool SetProperty<T>(ref T field, T value, [CallerMemberName] string propertyName = null)
+        {
+            if (!EqualityComparer<T>.Default.Equals(field, value))
+            {
+                field = value;
+                OnPropertyChanged(propertyName);
+                return true;
+            }
+            return false;
+        }
+
+        private void DocumentPathBrowse_Click(object sender, RoutedEventArgs e)
+        {
+            // TODO: Review use of System.Windows.Forms reference.
+            var dialog = new System.Windows.Forms.FolderBrowserDialog();
```

Comment:
The OpenFileDialog in WPF requires you to select a file and won't let you select a folder reference. The two workaround I see posted more frequently use one of two methods:

* System.Windows.Forms.FolderBrowserDialog from the .NET framework --> https://msdn.microsoft.com/en-us/library/system.windows.forms.folderbrowserdialog.aspx
* CommonOpenFileDialog from the Windows API Code pack --> https://github.com/aybe/Windows-API-Code-Pack-1.1

Did you have a preference or know of a pure WPF way to implement the select a folder dialog?

I'm not a WPF developer (one reason I wanted to contribute to this project was to learn some WPF) -- sorry if this seems simplistic.